Add guide for Azure Postgres/MySQL database access (#9729)
gravitational/teleport: docs/pages/database-access/guides/azure-postgres-mysql.mdx

  Attach this identity to all Azure VMs that will be running Database Service
  agents.
</TabItem>
<TabItem label="Using app registrations">
  <Admonition type="note">
    Registering Database Service as Azure AD application is suitable for test
    and development scenarios, or if your Database Service does not run on an
    Azure VM. For production scenarios prefer to use the managed identity
    approach.
  </Admonition>

  Go the the [App registrations](https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredApps)
  page of your Azure Active Directory and click on *New registration*:

  ![App registrations](../../../img/database-access/guides/azure/app-registrations@2x.png)

  Pick a name (e.g. _DatabaseService_) and register a new application. Once the
  app has been created, take note of its _Application (client) ID_ and click on
  *Add a certificate or secret*:

  ![Registered app](../../../img/database-access/guides/azure/registered-app@2x.png)

  Create a new client secret that Database Service agent will use to
  authenticate with Azure API:

  ![Registered app secrets](../../../img/database-access/guides/azure/registered-app-secrets@2x.png)

>   Teleport Database Service uses Azure SDK default credential provider chain to

Reviewer: ```suggestion
  Teleport Database Service uses Azure SDK's default credential provider chain to
```

  look for credentials. Refer to [Authentication methods](https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authorization)
  to pick a method suitable for your use-case. For example, to use

>   environment-based authentication with client secret, Database Service should

Reviewer: ```suggestion
  environment-based authentication with a client secret, Database Service should
```

  have the following environment variables set:

  ```bash
  export AZURE_TENANT_ID=
  export AZURE_CLIENT_ID=
  export AZURE_CLIENT_SECRET=
  ```
</TabItem>
</Tabs>

## Step 3/4. Create Azure database users

To let Teleport connect to your Azure database authenticating as a service
principal you need to create Azure AD users for that principal in the database.

### Assign Azure AD administrator

> Only Azure AD administrator for the database can connect to it and create

Reviewer: ```suggestion
Only the Azure AD administrator for the database can connect to it and create
```

> subsequent Azure AD users. Go to your database _Active Directory admin_ page

Reviewer: ```suggestion
Azure AD users. Go to your database's _Active Directory admin_ page
```

> and set the AD admin using _Set admin_ button:

Reviewer: ```suggestion
and set the AD admin using the _Set admin_ button:
```

![Set AD admin](../../../img/database-access/guides/azure/set-ad-admin@2x.png)

Note that only one AD user can be set as an admin for the database.

> ### Connect to database as AD admin

Reviewer: ```suggestion
### Connect to the database as an AD admin
```

Next, you need to connect to your database as the AD admin user.

> Use Azure `az` CLI utility to log in as the user that you set as the AD admin,

Reviewer: ```suggestion
Use the Azure `az` CLI utility to log in as the user that you set as the AD admin,
```

fetch the access token and use it as a password when connecting to the database:

<Tabs>
<TabItem label="PostgreSQL">
  ```bash
  $ az login -u ad@example.com
  $ export PGPASSWORD=`az account get-access-token --resource-type oss-rdbms --output tsv --query accessToken`
  $ psql "host=example.postgres.database.azure.com user=ad@example.com@instance-name sslmode=require dbname=postgres"
  ```
</TabItem>
<TabItem label="MySQL">
  ```bash
  $ az login -u ad-admin@example.com
  $ export TOKEN=`az account get-access-token --resource-type oss-rdbms --output tsv --query accessToken`
  $ mysql -h example.mysql.database.azure.com -P 3306 -u ad@example.com@instance-name --enable-cleartext-plugin --password=$TOKEN
  ```
</TabItem>
</Tabs>

Note that the database username must include `@instance-name` suffix with the
name of the Azure database instance you're connecting to.

### Create AD users

Once connected to the database as AD admin, create database users for the

> service principal Teleport Database Service will be using. Use `Client ID`

Reviewer: ```suggestion
service principal that Teleport Database Service will be using. Use `Client ID`
```

when using managed identities and `Application (client) ID` when using app
registrations:

<Tabs>
<TabItem label="PostgreSQL">
  ```
  postgres=> SET aad_validate_oids_in_tenant = off;
  SET
  postgres=> CREATE ROLE teleport WITH LOGIN PASSWORD '11111111-2222-3333-4444-555555555555' IN ROLE azure_ad_user;
  CREATE ROLE
  ```
</TabItem>
<TabItem label="MySQL">
  ```
  mysql> SET aad_auth_validate_oids_in_tenant = OFF;
  mysql> CREATE AADUSER 'teleport' IDENTIFIED BY '11111111-2222-3333-4444-555555555555';
  Query OK, 0 rows affected (0.92 sec)
  ```
</TabItem>
</Tabs>

You can create multiple database users for the same service principal.

## Step 4/4. Connect

> Log into your Teleport cluster, your Azure database should appear in the list of

Reviewer: ```suggestion
Log into your Teleport cluster. Your Azure database should appear in the list of
```

available databases:

```code
$ tsh login --proxy=teleport.example.com --user=alice
$ tsh db ls
# Name     Description         Labels
# -------- ------------------- -------
# azure-db                     env=dev
```

Fetch short-lived client certificate for it using `tsh db login` command:

```code
$ tsh db login --db-user=teleport azure-db
```

<Admonition type="tip">
  You can be logged into multiple databases simultaneously.
</Admonition>

Now connect to the database:

```code
$ tsh db connect azure-db
```

<Admonition type="note">
  The appropriate database command-line client (`psql`, `mysql`) should be

>   available in PATH in order to be able to connect.

Reviewer: Does this refer to the end-user's machine or the VM running the Database Service?
